Database Generation Schema

Teg64 opened the pull request:
SQL script for initial generation/setup of the MySQL database server.

[Related UML diagram](https://lucid.app/lucidchart/f8be74e2-f0ba-4d04-9f81-aaeb75e86f65/edit?page=-d_cnpzMqLxT&invitationId=inv_55430a68-14f9-40c1-8e46-8e4b16ea5d20#)

Teg64 committed:
Test commit

Teg64 committed:
Created initial DB generation schema

Schema includes all tables and their primary + foreign key constraints

Teg64 committed:
Changed upload_datetime to be a DATETIME instead of a DATE

JMSS-Unknown reviewed (changes requested):
Looks good, some formatting inconsistencies on lines 26, 70 & 71. Should we introduce a linter?

Teg64 committed:
Made minor alteration to DB schema

Removed username from rabit_user and made display_name and email mandatory. Additionally, completely reformatted all code and added a simple test script

Teg64 commented:
> Looks good, some formatting inconsistencies on lines 26, 70 & 71. Should we introduce a linter?

Just made some formatting changes. The previous issues were a result of the (not great) formatter used by MySQL workbench. I reformatted everything using Datagrip, so it should be better and more consistent now

JMSS-Unknown reviewed (approved):
Looks good
